File: appMultiMedia/src/main/java/com/au/audiorecordplayer/recorder/IWaveDetectRecord.java
package com.au.audiorecordplayer.recorder;

public interface IWaveDetectRecord {
    interface IWaveDetectCallback {
        /**
         * @param rms 返回的是16位PCM数据的RMS
         * @param db 大致能看出一般是-50左右，说话后是-20～-10大声。
         */
        void onWaveDetect(double rms, double db);
    }

    void setWaveDetectCallback(IWaveDetectCallback callback);
}

File: appMultiMedia/src/main/java/com/au/audiorecordplayer/recorder/IWaveDetectRecord.kt
package com.au.audiorecordplayer.recorder

interface IWaveDetectRecord {
    interface IWaveDetectCallback {
        /**
         * @param db 大致能看出一般是-50左右，说话后是-20～-10大声。
         */
        fun onWaveDetect(db: Double)
    }

    fun setWaveDetectCallback(callback: IWaveDetectCallback?)
    fun processAudioData(buffer: ByteArray?, readBytes: Int)
}